warden/jwt.go

```diff
@@ -0,0 +1,63 @@
+package warden
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	auth0 "github.com/auth0-community/go-auth0"
+	"github.com/gin-gonic/gin"
+	jose "gopkg.in/square/go-jose.v2"
+	jwt "gopkg.in/square/go-jose.v2/jwt"
+)
+
+func verifyJWT(request *http.Request) (*jwt.Claims, error) {
+	domain := request.Header.Get("Auth0-Domain")
+	if domain == "" {
+		return nil, errors.New("Auth0-Domain header missing")
+	}
+
+	apiIdentifer := request.Header.Get("Auth0-Audience")
+	if apiIdentifer == "" {
+		return nil, errors.New("Auth0-API-Identifier header missing")
+	}
+
+	jwksURI := fmt.Sprintf("https://%s.auth0.com/.well-known/jwks.json", domain)
+	apiIssuer := fmt.Sprintf("https://%s.auth0.com/", domain)
```

Comment:
I don't think this should be specific to auth0 but most likely specific to OpenID Connect IDP, we should probably also use the config to store the domain including ``auth0.com``